core/Tracker/Db/Pdo/Mysql.php

```diff
@@ -260,10 +260,7 @@ public function lastInsertId()
      */
     public function isErrNo($e, $errno)
     {
-        if (preg_match('/([0-9]{4})/', $e->getMessage(), $match)) {
-            return $match[1] == $errno;
-        }
-        return false;
+        return \Piwik\Db\Adapter\Pdo\Mysql::isPdoErrorNumber($e, $errno);
```

Comment:
Does this mean the `\Piwik\Db\Adapter\Pdo\...` classes need to be required in `/piwik.php` now?

Reply:
will merge now but @tsteur if this needs an update please go ahead and merge it in a separate PR

Reply:
@diosmosis not sure what you mean? The files will be auto loaded if that's what you mean?

Reply:
I thought autoloading wasn't used in the tracker side for performance (hence the `require_once`s in `PIWIK_INCLUDE_PATH/piwik.php`). I guess it is used?

Reply:
It is definitely used as it's needed for all the plugins etc. those requires could likely be removed.